torchvision/models/detection/roi_heads.py

```diff
@@ -603,9 +603,7 @@ def postprocess_detections(self, class_logits, box_regression, proposals, image_
             boxes, scores, labels = boxes[keep], scores[keep], labels[keep]
 
             # non-maximum suppression, independently done per class
-            keep = box_ops.batched_nms(boxes, scores, labels, self.nms_thresh)
-            # keep only topk scoring predictions
-            keep = keep[:self.detections_per_img]
+            keep = box_ops.batched_nms(boxes, scores, labels, self.nms_thresh, self.detections_per_img)
```

Comment:
Is `len(boxes) >= self.detections_per_img` pretty much guaranteed at training time? Wondering since we have added a check in the nms C++ implementation for it.